packages/nc-gui/components/smartsheet/column/QrCodeOptions.vue

```diff
@@ -0,0 +1,73 @@
+<script setup lang="ts">
+import { UITypes } from 'nocodb-sdk'
+import type { SelectProps } from 'ant-design-vue'
+import { useVModel } from '#imports'
+
+const props = defineProps<{
+  value: any
+}>()
+const emit = defineEmits(['update:value'])
+
+const meta = inject(MetaInj, ref())
+
+const activeView = inject(ActiveViewInj, ref())
+
+const reloadDataHook = inject(ReloadViewDataHookInj)!
+
+const { fields, metaColumnById } = useViewColumns(activeView, meta, () => reloadDataHook.trigger())
+
+const vModel = useVModel(props, 'value', emit)
+
+const { setAdditionalValidations, validateInfos, column } = useColumnCreateStoreOrThrow()
+
+const allowedColumnTypesForQrValue = [
+  UITypes.Formula,
+  UITypes.SingleLineText,
+  UITypes.LongText,
+  UITypes.PhoneNumber,
+  UITypes.URL,
+  UITypes.Email,
+] as string[]
+
+const columnsAllowedAsQrValue = computed<SelectProps['options']>(() => {
+  return fields.value
+    ?.filter((el) => el.fk_column_id && allowedColumnTypesForQrValue.includes(metaColumnById.value[el.fk_column_id].uidt))
+    .map((field) => {
+      return {
+        value: field.fk_column_id,
+        label: field.title,
+      }
+    })
+})
+
+// set default value
+vModel.value.fk_qr_value_column_id = (column?.value?.colOptions as Record<string, any>)?.fk_qr_value_column_id || ''
```

Comment:
This will only be executed once (on setup) and never re-executed (as long as the component is not unmounted and re-mounted)
Is this intended?

Reply:
I found this in several other colum Options components - so I thought I stay consistent, even though if it's maybe not idiomatic regarding life cycle management, and if refactoring the Options component, then all together (and in alignment with someone with already more vue and nocodb FE expertise :) )
I've put it now though into a onMounted call